Core/WebCacheManager.swift

```diff
@@ -22,45 +22,57 @@ import WebKit
 
 public class WebCacheManager {
 
+    public static var instance = WebCacheManager()
+    
     private struct Constants {
-        static let internalCache = "duckduckgo.com"
+        static let cookieDomain = "duckduckgo.com"
     }
     
-    private static var allData: Set<String> {
+    private var cookies = [HTTPCookie]()
+    
+    private var allDataTypes: Set<String> {
         return WKWebsiteDataStore.allWebsiteDataTypes()
     }
     
-    private static var dataStore: WKWebsiteDataStore {
+    private var dataStore: WKWebsiteDataStore {
         return WKWebsiteDataStore.default()
     }
     
-    /**
-     Provides a summary of the external (non-duckduckgo) cached data
-     */
-    public static func summary(completionHandler: @escaping (_ summary: WebCacheSummary) -> Void) {
-         dataStore.fetchDataRecords(ofTypes: allData, completionHandler: { records in
-            let count = records.reduce(0) { (count, record) in
-                if record.displayName == Constants.internalCache {
-                    return count
-                }
-                return count + record.dataTypes.count
+    private init() {
+        self.cookies = HTTPCookieStorage.shared.cookies ?? []
+    }
+
+    public func injectCookies(dataStore: WKWebsiteDataStore) {
+        
+        if #available(iOS 11, *) {
+            for cookie in cookies {
+                dataStore.httpCookieStore.setCookie(cookie)
             }
-            Logger.log(text: String(format: "Web cache retrieved, there are %d items in the cache", count))
-            completionHandler(WebCacheSummary(count: count))
-        })
+        }
     }
     
     /**
-     Clears the cache of all external (non-duckduckgo) data
+     Clears the cache of all data, except duckduckgo cookies
      */
-    public static func clear(completionHandler: @escaping () -> Void) {
-        dataStore.fetchDataRecords(ofTypes: allData) { records in
-            let externalRecords = records.filter { $0.displayName != Constants.internalCache }
-            dataStore.removeData(ofTypes: allData, for: externalRecords) {
-                Logger.log(text: "External cache cleared")
-                completionHandler()
+    public func clear() {
+        if #available(iOS 11, *) {
+            extractAllowedCookies(in: dataStore.httpCookieStore)
```

Comment:
I think this could be problematic because we move straight onto the clearing before the completion in `extractAllowedCookies` method is done

Reply:
Good spot - I've updated the tests above to cover this as well.